sources/packages/api/src/route-controllers/designation-agreement/designation-agreement.aest.controller.ts

```diff
@@ -0,0 +1,136 @@
+import {
+  Controller,
+  Get,
+  Param,
+  NotFoundException,
+  Patch,
+  Body,
+  Query,
+  BadRequestException,
+} from "@nestjs/common";
+import { DesignationAgreementService, FormService } from "../../services";
+import { FormNames } from "../../services/form/constants";
+import { DesignationAgreementStatus } from "../../database/entities";
+import { getISODateOnlyString, PaginationParams } from "../../utilities";
+import { AuthorizedParties } from "../../auth/authorized-parties.enum";
+import { AllowAuthorizedParty, Groups, UserToken } from "../../auth/decorators";
+import { IUserToken } from "../../auth/userToken.interface";
+import { UserGroups } from "../../auth/user-groups.enum";
+import {
+  GetDesignationAgreementDto,
+  GetDesignationAgreementsDto,
+  PendingDesignationDto,
+  UpdateDesignationDto,
+} from "./models/designation-agreement.model";
+import { DesignationAgreementServiceController } from "./designation-agreement.service.controller";
+
+@AllowAuthorizedParty(AuthorizedParties.aest)
+@Groups(UserGroups.AESTUser)
+@Controller("designation-agreement")
+export class DesignationAgreementAESTController {
+  constructor(
+    private readonly designationAgreementServiceController: DesignationAgreementServiceController,
+    private readonly designationAgreementService: DesignationAgreementService,
+    private readonly formService: FormService,
+  ) {}
+
+  /**
+   * Retrieve the designation agreement information and
+   * the associated locations approvals.
+   * @param designationId
+   * @returns designation agreement information.
+   */
+  @Get(":designationId")
+  async getDesignationAgreement(
+    @Param("designationId") designationId: number,
+  ): Promise<GetDesignationAgreementDto> {
+    return this.designationAgreementServiceController.getDesignationAgreement(
+      designationId,
+    );
+  }
+
+  /**
+   * Get the list of all the designations that belongs to
+   * the institution.
+   * @param institutionId
+   * @returns the list of all the designations that
+   * belongs to the institution.
+   */
+  @Get("institution/:institutionId")
+  async getDesignationAgreements(
+    @Param("institutionId") institutionId: number,
+  ): Promise<GetDesignationAgreementsDto[]> {
+    return this.designationAgreementServiceController.getDesignationAgreements(
+      institutionId,
+    );
+  }
+
+  /**
+   * API to retrieve all pending designations.
+   * @param designationStatus
+   * @param searchCriteria to search designation.
```

Comment:
mm, here are not using lazy pagination right. so, we can actually use the inbuild data table search option, right? instead of an API search. just a thought team @andrewsignori-aot @guru-aot @henryoforeh-aot @dheepak-aot

Reply:
When we are not using the lazy loading and the built-in search is enough to achieve the requirement I do not see a reason to not use it.

Reply:
Maybe we can sync up to discuss this tomorrow.

Reply:
The PrimeVue data table requires some coding to enable the OOTB global search. 
https://github.com/primefaces/primevue/blob/master/src/views/datatable/DataTableFilterDemo.vue
Hence primevue is a temporary solution, we must consider enabling OOTB search on non-lazy data fetch when moving to Vuetify3.

Reply:
maybe you can add a todo to remove the API change when vuetify data-table search is implemented

Reply:
IMO, I would like to add a todo at Vue component and leave API with the search as it is an optional parameter.